src/metabase/query_processor/middleware/resolve_fields.clj

```diff
@@ -9,8 +10,13 @@
 
 (defn- resolve-fields-with-ids!
   [field-ids]
-  (qp.store/fetch-and-store-fields! field-ids)
-  (when-let [parent-ids (seq (filter some? (map (comp :parent_id qp.store/field) field-ids)))]
+  (qp.store/bulk-metadata :metadata/column field-ids)
+  (when-let [parent-ids (not-empty
+                         (into []
```

Comment:
I feel like I want a `into-not-empty` util, I think there are other places in the code this same pattern happens.